data-prepper-plugins/aggregate-processor/src/main/java/com/amazon/dataprepper/plugins/processor/aggregate/AggregateActionResponse.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.amazon.dataprepper.plugins.processor.aggregate;
+
+import com.amazon.dataprepper.model.event.Event;
+
+import java.util.Optional;
+
+/**
+ * Model class to be returned in {@link com.amazon.dataprepper.plugins.processor.aggregate.AggregateAction}. Contains both the Event to be processed and an option
+ * to close the current window for {@link com.amazon.dataprepper.plugins.processor.aggregate.AggregateProcessor} immediately after an event is handled.
+ * @since 1.3
+ */
+public class AggregateActionResponse {
+    private Optional<Event> event;
```

Comment:
Optional was mainly intended for return values from methods, and not for properties of Java classes. We should consider replacing with with Event type. Also Optional is not serializable, so if we plan to serialize AggregateActionResponse object we will get unexpected data structure in the final output.